nav2_waypoint_follower/plugins/wait_at_waypoint.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 Fetullah Atas
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <pluginlib/class_list_macros.hpp>
+#include <string>
+#include <exception>
+
+#include "nav2_util/node_utils.hpp"
+#include "nav2_waypoint_follower/plugins/wait_at_waypoint.hpp"
+
+namespace nav2_waypoint_follower
+{
+WaitAtWaypoint::WaitAtWaypoint()
+: waypoint_pause_duration_(0),
+  is_enabled_(true)
+{
+}
+
+WaitAtWaypoint::~WaitAtWaypoint()
+{
+}
+
+void WaitAtWaypoint::initialize(
+  const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
+  const std::string & plugin_name)
+{
+  auto node = parent.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node  in wait at waypoint plugin!"};
+  }
+  logger_ = node->get_logger();
+  nav2_util::declare_parameter_if_not_declared(
+    node,
+    plugin_name + ".waypoint_pause_duration",
+    rclcpp::ParameterValue(0));
+  nav2_util::declare_parameter_if_not_declared(
+    node,
+    plugin_name + ".enabled",
+    rclcpp::ParameterValue(false));
+  node->get_parameter(
+    plugin_name + ".waypoint_pause_duration",
+    waypoint_pause_duration_);
+  node->get_parameter(
+    plugin_name + ".enabled",
+    is_enabled_);
+  if (!waypoint_pause_duration_) {
```

Comment:
Check if 0, using a boolean operator on an int technically works, but that's really unclear and open for misinterpretation